### PR TITLE
feat(eks): Support KMS CMK on handler functions

### PR DESCRIPTION
**Pull request by cmckni3**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by cmckni3** [comment]
For my use case, need to allow the provider handlers to use KMS keys as well.

File: packages/@aws-cdk/aws-eks/lib/cluster-resource-provider.ts [hunk 1]
```diff
@@ -81,6 +87,7 @@ export class ClusterResourceProvider extends NestedStack {
       description: 'onEvent handler for EKS cluster resource provider',
       runtime: HANDLER_RUNTIME,
       environment: props.environment,
+      environmentEncryption: props.environmentEncryption,
```
Comment (cmckni3) [inline packages/@aws-cdk/aws-eks/lib/cluster-resource-provider.ts:90]: In my case, I would need a KMS key created in the nested stack as well.

**Comment by otaviomacedo** [comment]
Hi, @cmckni3. 

Thanks for the contribution. Can you please just fix the failing tests?

**Comment by cmckni3** [comment]
Reopen

**Comment by cmckni3** [comment]
> Hi, @cmckni3.
> 
> Thanks for the contribution. Can you please just fix the failing tests?

Sure, I'll work on fixing tests when I have a chance.

**Pull request closed without merging**